src/components/AddressSearch/index.tsx

```diff
@@ -97,26 +96,25 @@ function AddressSearch(
     const containerRef = useRef<View>(null);
     const query = useMemo(
         () => ({
-            language: preferredLocale,
-            types: resultTypes,
-            components: isLimitedToUSA ? 'country:us' : undefined,
-            ...(locationBias && {locationbias: locationBias}),
+            languageCode: preferredLocale,
+            includedRegionCodes: isLimitedToUSA ? ['us'] : undefined,
+            ...(locationBias && {locationBias}),
         }),
-        [preferredLocale, resultTypes, isLimitedToUSA, locationBias],
+        [preferredLocale, isLimitedToUSA, locationBias],
     );
     const shouldShowCurrentLocationButton = canUseCurrentLocation && searchValue.trim().length === 0 && isFocused;
     const saveLocationDetails = (autocompleteData: GooglePlaceData, details: GooglePlaceDetail | null) => {
-        const addressComponents = details?.address_components;
+        const addressComponents = details?.addressComponents;
         if (!addressComponents) {
             // When there are details, but no address_components, this indicates that some predefined options have been passed
             // to this component which don't match the usual properties coming from auto-complete. In that case, only a limited
             // amount of data massaging needs to happen for what the parent expects to get from this function.
             if (details) {
                 onPress?.({
                     address: autocompleteData.description ?? '',
-                    lat: details.geometry.location.lat ?? 0,
-                    lng: details.geometry.location.lng ?? 0,
-                    name: details.name,
+                    lat: details.location?.latitude ?? details.geometry.location?.lat ?? 0,
+                    lng: details.location?.longitude ?? details.geometry.location?.lng ?? 0,
+                    name: autocompleteData?.structured_formatting?.main_text ?? details?.name ?? '',
```

Comment:
`details.name` isn't the same as it was in the old API, the format here is `places/<id>` so not really good to use here. The next closest thing is `details?. shortFormattedAddress`

Reply:
In this context, the `details.name` is not used from old API. Instead, it indicates a predefined option i.e. saved places in our app. So, I think we need to keep 
 `details.name`

Reply:
> I don't understand what you mean, can you show me?

Please find the test video that demonstrates the use of `predefined` places. This is the reason why I think we need to keep `details.name`.


https://github.com/user-attachments/assets/0ad30e8b-90e9-42b5-809d-bb4cd712f533